Golem/ActivityLoop.cs

```diff
@@ -86,41 +82,80 @@ await foreach (var trackingEvent in _yagnaApi.ActivityMonitorStream(token))
         }
         finally
         {
-            _logger.LogInformation("Activity monitoring loop closed. Current job clenup");
-            _jobs.SetAllJobsFinished();
-            setCurrentJob(null);
+            _logger.LogInformation("Activity monitoring loop closed. Current job cleanup");
+            _jobs.CleanupJobs();
+            _jobs.SetCurrentJob(null);
+        }
+    }
+
+    public async Task AgreementsLoop()
+    {
+        _logger.LogInformation("Starting monitoring agreement events");
+
+        DateTime since = DateTime.Now;
+        while (true)
+        {
+            try
+            {
+                _token.ThrowIfCancellationRequested();
+                _logger.LogDebug("Checking for new Agreement events since: {}", since);
+
+                var events = await _yagnaApi.GetAgreementEvents(since, _token);
+                if (events.Count > 0)
+                {
+                    var agreements = FilterEvents(events);
+
+                    List<Job> currentJobs = await UpdateJobs(_jobs, agreements.ToList());
+                    _jobs.SetCurrentJob(SelectCurrentJob(currentJobs));
+
+                    since = events.Max(evt => evt.EventDate);
```

Comment:
this should be wrapped with finally b/c when an agreement is terminated then `UpdateJobs` throws an exception when a Rest call to agreement api gets 404 and then GetAgreementEvents retrieves the same event all the time making the UpdateJobs throw an exception again and again...
Not sure what should happen with `SetCurrentJob` though

e.g.:
```
try
{
    var agreements = FilterEvents(events);

    List<Job> currentJobs = await UpdateJobs(_jobs, agreements.ToList());
    _jobs.SetCurrentJob(SelectCurrentJob(currentJobs));
}
finally
{
      since = events.Max(evt => evt.EventDate);
}
```

Reply:
Similarly InvoiceEventsLoop could be updated